src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,313 @@
+import os
+import logging
+import numpy as np
+from time import sleep
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+try:
+    import clr
+
+    lf_root = os.environ['LIGHTFIELD_ROOT']
+    automation_path = lf_root + '\PrincetonInstruments.LightField.AutomationV4.dll'
+    addin_path = lf_root + '\AddInViews\PrincetonInstruments.LightFieldViewV4.dll'
+    support_path = lf_root + '\PrincetonInstruments.LightFieldAddInSupportServices.dll'
+
+    addin_class = clr.AddReference(addin_path)
+    automation_class = clr.AddReference(automation_path)
+    support_class = clr.AddReference(support_path)
+
+    import PrincetonInstruments.LightField as lf
+
+    clr.AddReference("System.Collections")
+    clr.AddReference("System.IO")
+    from System.Collections.Generic import List
+    from System import String
+    from System.IO import FileAccess
+except Exception as e:
+    logger.error(f"Exception occurred during import: {e}")
+
+
+class LightfieldApp:
+    def __init__(self, visible):
+
+        self._addinbase = lf.AddIns.AddInBase()
+        self._automation = lf.Automation.Automation(visible, List[String]())
+        self._application = self._automation.LightFieldApplication
+        self._experiment = self._application.Experiment
+
+    @property
+    def automation(self):
+        return self._automation
+
+    @property
+    def addinbase(self):
+        return self._addinbase
+
+    @property
+    def application(self):
+        return self._application
+
+    @property
+    def experiment(self):
+        return self._experiment
+
+    def __del__(self):
+        """
+        Uses Python garbage collection method used to terminate
+        AddInProcess.exe, if this is not done, LightField will not reopen
+        properly.
+        """
+        self.automation.Dispose()
+        logger.info('Closed AddInProcess.exe')
+
+    def set(self, setting, value):
```

Comment:
First point: If we make them more verbose we would just make each line that uses it a lot longer (and they are already pretty long. I think the names are fine as they are just helpers in a sense.

Second point: In its current state it is already general and can already set any of the settings available in the LF API as it just checks if that value is an existable one in Lightfield and sets it

Third point: Looking at the current setup the setting is not permanent unless you save it. I am not fully sure of this though...

Reply:
I can try make Lightfield either open up a fresh experiment with no name or fiddle with the datagenerator so that we can let people create an experiment with their own name and settings. The latter seems better for everyone